v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/utils/ElasticsearchIO.java

```diff
@@ -1513,21 +1541,31 @@ private HttpEntity handleRetry(
           String method, String endpoint, Map<String, String> params, HttpEntity requestBody)
           throws IOException, InterruptedException {
         Response response;
-        HttpEntity responseEntity;
+        HttpEntity responseEntity = null;
         Sleeper sleeper = Sleeper.DEFAULT;
         BackOff backoff = retryBackoff.backoff();
         int attempt = 0;
         // while retry policy exists
         while (BackOffUtils.next(sleeper, backoff)) {
-          LOG.warn(String.format(RETRY_ATTEMPT_LOG, ++attempt));
-          Request request = new Request(method, endpoint);
-          request.addParameters(params);
-          request.setEntity(requestBody);
-          response = restClient.performRequest(request);
-          responseEntity = new BufferedHttpEntity(response.getEntity());
+          LOG.warn(RETRY_ATTEMPT_LOG, ++attempt);
```

Comment:
I think removing String.format won't work, because `RETRY_ATTEMPT_LOG` has `%d` in the message as the placeholder, and `LOG.warn` expects `{ }` as the placeholder.

Reply:
sure, I had copied this from the beam version of ElasticsearchIO here: https://github.com/apache/beam/blob/master/sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java#L2547

I did that to make retry work properly as it looked like the logic in the beam ElasticsearchIO had functional retries whereas this version did not (at least not in conventional failure cases that would throw exceptions)

But yes, I can definitely add back the String.format

Reply:
Note that the Beam version has a proper `RETRY_ATTEMPT_LOG` message (with `{}`, not `%d`):
https://github.com/apache/beam/blob/bf39489b2a1fd45e6798483d083e4ad240f66891/sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java#L1981
So either use String.format with a `%d` message or direct logging with a `{}` message.